scipy/interpolate/_polyint.py

```diff
@@ -677,8 +753,94 @@ def _evaluate(self, x):
                 p[r[:-1]] = self.yi[r[-1]]
         return p
 
+    def derivative(self, x, der=1):
+        """
+        Evaluate a single derivative of the polynomial at the point x.
+
+        Parameters
+        ----------
+        x : array_like
+            Point or points at which to evaluate the derivatives
+        der : integer, optional
+            Which derivative to evaluate (default: first derivative).
+            This number includes the function value as 0th derivative.
+
+        Returns
+        -------
+        d : ndarray
+            Derivative interpolated at the x-points. Shape of `d` is
+            determined by replacing the interpolation axis in the
+            original array with the shape of `x`.
+        """
+        x, x_shape = self._prepare_x(x)
+        y = self._evaluate_derivatives(x, der+1, all_lower=False)
+        return self._finish_y(y, x_shape)
+
+    def _evaluate_derivatives(self, x, der=None, all_lower=True):
+        # NB: der here is not the order of the highest derivative;
```

Comment:
This is just the meaning of `der` within the existing `_Interpolator1DWithDerivatives` class, not something I've chosen or added. The implementation here has to match for the class methods to work.

I agree it's misleading, which is why I added the comment in case any

It could easily be renamed as this function isn't in the public API, but I'm not sure what the procedure is for that kind of thing.

Reply:
Ok if there is a precedent then it's fine to leave